Expensify/App
# Clear network queue when signing out

@cead22 will you review please?

https://github.com/Expensify/Expensify/issues/142726 seems to have fixed most of the issues with this already, but this is another safety measure to make sure we aren't spamming the api when we are on the signin screen

# Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/142252

### Tests
**I could not think of specific tests that would get us into a state where we would have normally been spamming requests. If you know of any way to reproduce that, let me know**

1. Sign in
2. Send a message
3. Sign out
4. Sign in on another platform, verify the message was sent & received successfully.


### Screenshots
n/a
